components/helm-broker/internal/addon/provider/git.go

```diff
@@ -60,18 +76,46 @@ func (g *GitGetter) IndexReader() (io.ReadCloser, error) {
 }
 
 // AddonLoadInfo returns information how to load addon
-// TODO: add feature for uploading documentation: https://github.com/kyma-project/kyma/issues/5040
 func (g *GitGetter) AddonLoadInfo(name addon.Name, version addon.Version) (LoadType, string, error) {
 	var (
 		addonDirName = fmt.Sprintf("%s-%s", name, version)
 		pathToAddon  = path.Join(g.dst, g.addonDirPath, addonDirName)
+		pathToDocs   = path.Join(pathToAddon, "/docs")
+		pathToTgz    = fmt.Sprintf("%s/docs-%s.tgz", g.tmpDir, addonDirName)
 	)
 
+	if _, err := os.Stat(pathToDocs); os.IsNotExist(err) {
+		return DirectoryLoadType, pathToAddon, err
+	}
+
+	err := archiver.Archive([]string{pathToDocs}, pathToTgz)
+	if err != nil {
+		return DirectoryLoadType, pathToAddon, exerr.Wrapf(err, "while creating archive '%s'", pathToTgz)
+	}
+
+	file, err := os.Open(pathToTgz)
+	if err != nil {
+		return DirectoryLoadType, pathToAddon, err
+	}
+	defer func() {
+		os.Remove(pathToTgz)
+		file.Close()
+	}()
+
+	docs, err := ioutil.ReadAll(file)
+	if err != nil {
+		return DirectoryLoadType, pathToAddon, err
```

Comment:
missing err wrap, same below, 

in case of error use `UnknownLoadType` and empty string instead of `DirectoryLoadType, pathToAddon`